packages/gatsby/src/commands/build.js

```diff
@@ -33,10 +33,14 @@ const waitJobsFinished = () =>
       if (store.getState().jobs.active.length === 0) {
         resolve()
         emitter.off(`END_JOB`, onEndJob)
+        return true
       }
+      return false
+    }
+    if (!onEndJob()) {
+      console.log(`[gatsby] waiting for jobs to finish`)
+      emitter.on(`END_JOB`, onEndJob)
```

Comment:
changes here are just for debugging purpose - they will need to be removed later